src/tx/applyTx/checkSpendCond.js

```diff
@@ -227,7 +227,8 @@ module.exports = async (state, tx, bridgeState, nodeConfig = {}) => {
       // we only allow one spending condition in an transaction, do we want to throw if we find more?
       spendingInput = input;
       spendingInputUnspent = unspent;
-      spendingAddrBuf = utils.ripemd160(spendingInput.script);
+      const { script } = spendingInput;
+      spendingAddrBuf = utils.ripemd160(script.length, utils.keccak256(script));
```

Comment:
How does `ethereumjs-util` handle the int in `ripemd`, does it pad it somehow to a fixed size integer?

Reply:
🤦‍♂ sorry, it is all broken .. ripemd doesn't even accepts multiple args as inputs. Will fix this mess tomorrow

Reply:
should be really something like `ripemd160([keccak256(length), keccak256(script)])`

Should we pad the ripemd hash itself as well?

Reply:
Here is the current implementation: https://github.com/leapdao/leap-core/pull/114/files#diff-7e1ea5f3cb35910ce52c225628d785d7R4

Reply:
> Should we pad the ripemd hash itself as well? Though it will always be padded with 12 zero bytes

No need to pad the ripemd hash.

> Here is the current implementation: https://github.com/leapdao/leap-core/pull/114/files#diff-7e1ea5f3cb35910ce52c225628d785d7R4

We can just pad `script.length` with zero-bytes on the left side to 32 bytes like
`0x${script.length.toString(16).padStart(64, '0')}`
That way we have a constant-size variable and everything is nice :)